### PR TITLE
[gh-1554] upgrade rust version and fix syntax errors.

### LINKED ISSUE #1554: [Build] using asdf rust plugin to build rust toolchain results in incompatible errors.
Build process:

```zsh
cargo build && cp target/debug/rooch ~/.cargo/bin/
info: syncing channel updates for '1.74.0-aarch64-apple-darwin'
info: latest update on 2023-11-16, rust version 1.74.0 (79e9716c9 2023-11-13)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustfmt'
error: the 'cargo' binary, normally provided by the 'cargo' component, is not applicable to the '1.74.0-aarch64-apple-darwin' toolchain
```

I'm using [asdf-rust](https://github.com/code-lever/asdf-rust) plugin to manage rust versions and install pkgs.

The output shows there's no `1.74.0-aarch64-apple-darwin` for the cargo binary in the toolchain, whereas in the toolchain file:

```toml
1.74.0
```

The error occurs because its syntax is incorrect.

The rust-toolchain file is a TOML file format consisting of a format like below:

```toml
[toolchain]
channel = "nightly-2020-07-10"
components = [ "rustfmt", "rustc-dev" ]
targets = [ "wasm32-unknown-unknown", "thumbv2-none-eabi" ]
profile = "minimal"
```

The correct way, there should be an indicator and date for the version instead of the version number itself for the whole rust binaries.

Ref:
1. https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file
2. https://rust-lang.github.io/rustup/concepts/toolchains.html
3. https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html

### PR DESCRIPTION
**Pull request by feliciss**
## Summary

Upgrade rust version and fix errors.

- Closes #1554

### CODE REVIEW

**Comment by feliciss** [comment]
Help debug GitHub CI. @jolestar @popcnt-subodhi @yourmoonlight

**Comment by jolestar** [comment]
It seems like Rustup does not support it. 

```
profile = "minimal"
channel = "stable-2023-11-16" # https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html --component rustfmt --component clippy
error: error: invalid value '[toolchain]
profile = "minimal"
channel = "stable-2023-11-16" # https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html' for '<toolchain>...': invalid toolchain name: '[toolchain]
profile = "minimal"
channel = "stable-2023-11-16" # https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html'

For more information, try '--help'.
: invalid toolchain name: '[toolchain]
profile = "minimal"
channel = "stable-2023-11-16" # https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html'
Error: The process '/home/runner/.cargo/bin/rustup' failed with exit code 1
```

**Comment by popcnt-subodhi** [comment]
> Help debug GitHub CI. @jolestar @popcnt-subodhi @yourmoonlight

Will work on it soon.

@jolestar Let me figure it out, take a break

**Comment by popcnt1** [comment]
@feliciss 

try to remove inputs in .github/actions/rust-setup/action.yaml in action-rs(https://github.com/marketplace/actions/rust-toolchain)

reason:

CI rely on actions-rs/toolchain to install rust toolchain, and " Input has higher priority, so if you are want to use toolchain file, you need to remove the input from the workflow file."

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1,3 @@
-1.74.0
\ No newline at end of file
+[toolchain]
+profile = "minimal"
```
Comment (popcnt1) [inline rust-toolchain:2]: default here

we need clippy and rustfmt

**Review by popcnt1 — CHANGES_REQUESTED** [review]
1. use default profile to install toolchain
2. fix input in rust-setup/action.yaml

**Comment by feliciss** [comment]
> 1. default profile to install toolchain
> 2. fix input in rust-setup/action.yaml
> 3. try 1.75.0? (it's a good time to upgrade rust)

I tried to remove action.yaml but there still has issue.

**Comment by popcnt1** [comment]
> > 1. default profile to install toolchain
> > 2. fix input in rust-setup/action.yaml
> > 3. try 1.75.0? (it's a good time to upgrade rust)
> 
> I tried to remove action.yaml but there still has issue.

1. check_build_test(which you failed workflow) run on main branch (you could check it in the source code)
2. you could create a new PR to just modify .github/actions/rust-setup/action.yaml, and keep 1.74.0  firstly for checking new toolchain config works

File: .github/actions/rust-setup/action.yaml [hunk 2]
```diff
@@ -9,11 +9,6 @@ runs:
     - run: sudo apt-get update && sudo apt-get install build-essential ca-certificates clang curl git libpq-dev libssl-dev pkg-config lsof lld --no-install-recommends --assume-yes
       shell: bash
 
-    - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # pin@v1
```
Comment (popcnt1) [inline .github/actions/rust-setup/action.yaml:12]: just remove with is enough.

Keep uses: <arg>

Comment (feliciss) [reply .github/actions/rust-setup/action.yaml:12]: Resolved.

**Comment by feliciss** [comment]
> > > 1. default profile to install toolchain
> > > 2. fix input in rust-setup/action.yaml
> > > 3. try 1.75.0? (it's a good time to upgrade rust)
> > 
> > 
> > I tried to remove action.yaml but there still has issue.
> 
> 1. check_build_test(which you failed workflow) run on main branch (you could check it in the source code)
> 2. you could create a new PR to just modify .github/actions/rust-setup/action.yaml, and keep 1.74.0  firstly for checking new toolchain config works. Then another PR upgrade the rust version

I made this PR to just upgrading rust version. The rust-toolchain file format remains the same to be resolved.

File: .github/actions/rust-setup/action.yaml [hunk 3]
```diff
@@ -11,8 +11,8 @@ runs:
 
     - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # pin@v1
       with:
-        override: true
-        components: rustfmt, clippy
+          override: true
```
[the remark this answers is not in the document]
Comment (feliciss) [reply .github/actions/rust-setup/action.yaml:14]: > I wonder why add a tab here?

It's a copy-pasted mistake from the GitHub review system. Will fix in the next commit.

**Review by jolestar — APPROVED** [review]

**Pull request merged**